share/ansible_navigator/utils/catalog_collections.py

```diff
@@ -115,62 +114,57 @@ def _generate_checksum(file_path: str, relative_path: str) -> Dict:
     def _process_plugin_dir(
         self,
         plugin_type: str,
-        filenames: List,
-        file_checksums: Dict,
-        dirpath: str,
+        filenames: Generator[Path, None, None],
+        file_checksums: Dict[str, Dict],
         collection: Dict,
     ) -> None:
-        # pylint: disable=too-many-arguments
         """Process each plugin within one plugin directory.
 
         :param plugin_type: The type of plugins
         :param filenames: The filenames of the plugins
         :param file_checksums: The checksums for the plugin files
-        :param dirpath: The path of the directory containing the plugins
         :param collection: The details of the collection
         """
         for filename in filenames:
-            file_path = f"{dirpath}/{filename}"
-            relative_path = file_path.replace(collection["path"], "")
-            _basename, extension = os.path.splitext(filename)
-            if not filename.startswith("__") and extension == ".py":
-                checksum_dict = file_checksums.get(relative_path)
+            relative_path = Path(filename).relative_to(collection["path"])
+            if not str(filename).startswith("__"):
```

Comment:
Readability nit, invert this conditional and continue early

```python
if str(filename).startswith("__"):
    continue
```

Then the rest of the loop body can be outdented one level.